api/src/main/java/edu/wpi/first/shuffleboard/api/plugin/Plugin.java

```diff
@@ -64,9 +64,15 @@ protected Plugin() {
 
     this.groupId = description.group();
     this.name = description.name();
-    this.version = description.version();
-    this.description = description.summary();
-    this.artifact = new PluginArtifact(this.groupId, this.name, this.version);
+    this.version = Version.parse(description.version());
+    this.summary = description.summary();
+
+    if (groupId.contains(":")) {
+      throw new IllegalStateException("A plugin's group ID cannot contain a colon (':') character - " + groupId);
```

Comment:
`IllegalArgumentException`. This is not state.

Reply:
It's state set by the annotation. It's not `IllegalArgument` because there is no argument passed to the constructor

Reply:
The user is passing an argument into the annotation.